x-pack/plugins/security_solution/public/dashboards/components/dashboard_renderer.tsx

```diff
@@ -73,20 +103,28 @@ const DashboardRendererComponent = ({
     dashboardContainer?.updateInput({ timeRange, query, filters });
   }, [dashboardContainer, filters, query, timeRange]);
 
-  const handleDashboardLoaded = useCallback(
-    (container: DashboardAPI) => {
-      setDashboardContainer(container);
-      onDashboardContainerLoaded?.(container);
-    },
-    [onDashboardContainerLoaded]
-  );
-  return savedObjectId && canReadDashboard ? (
-    <DashboardContainerRenderer
-      ref={handleDashboardLoaded}
-      savedObjectId={savedObjectId}
-      getCreationOptions={getCreationOptions}
-    />
-  ) : null;
+  useEffect(() => {
+    if (isCreateDashboard && firstSecurityTagId)
+      dashboardContainer?.updateInput({ tags: [firstSecurityTagId] });
+  }, [dashboardContainer, firstSecurityTagId, isCreateDashboard]);
+
+  useEffect(() => {
+    if (wrapperRef) {
+      setDashboardContainerRenderer(
+        <DashboardContainerRenderer
+          ref={onDashboardContainerLoaded}
+          savedObjectId={savedObjectId}
+          getCreationOptions={getCreationOptions}
+        />
+      );
+    }
+
+    return () => {
+      setDashboardContainerRenderer(undefined);
+    };
+  }, [getCreationOptions, onDashboardContainerLoaded, refetchByForceRefresh, savedObjectId]);
+
+  return canReadDashboard ? <div ref={wrapperRef}>{dashboardContainerRenderer}</div> : null;
```

Comment:
Using a state to store a React Element is a bit odd. Can't we just do:
```suggestion
 return canReadDashboard ? (
    <div ref={wrapperRef}>
      {wrapperRef.current && (
        <DashboardContainerRenderer
          ref={onDashboardContainerLoaded}
          savedObjectId={savedObjectId}
          getCreationOptions={getCreationOptions}
        />
      )}
    </div>
  ) : null;
```

Reply:
I agree it looks not natural implementing it like this, but it wouldn't render if I don't set dashboardContainerRenderer in state...